guides/data.livemd

```diff
@@ -0,0 +1,265 @@
+
+```elixir
+# A simple bar plot using the shorthand api
+Data.chart(data, :bar, x: "category", y: "score")
+```
+
+```elixir
+# The same chart without the shorthand api
+Vl.new()
+|> Vl.data_from_values(data)
+|> Vl.mark(:bar)
+|> Vl.encode_field(:y, "score", type: :quantitative)
+|> Vl.encode_field(:x, "category", type: :nominal)
+```
+
+Plotting a simple chart is a breeze! As we can see from the comparison above, the code becomes much leaner and handleable. However, the API also accepts a list of options for each argument, allowing more complex results.
+
+```elixir
+# A line plot with point: true without the shorthand api
+Vl.new()
+|> Vl.data_from_values(fuels, only: ["total", "solid_fuel"])
+|> Vl.mark(:line, point: true)
+|> Vl.encode_field(:x, "total", type: :quantitative)
+|> Vl.encode_field(:y, "solid_fuel", type: :quantitative)
+```
+
+```elixir
+# A line plot with point: true using the shorthand api
+Data.chart(fuels, [type: :line, point: true], x: "total", y: "solid_fuel")
+```
+
+Now let's see a bit of iteroperability between the api and the main module. We'll plot the same line chart but now with a title and a custom width.
+
+```elixir
+# Without the shorthand api
+Vl.new(title: "Fuels", width: 400)
+|> Vl.data_from_values(fuels, only: ["total", "solid_fuel"])
+|> Vl.mark(:line, point: true)
+|> Vl.encode_field(:x, "total", type: :quantitative)
+|> Vl.encode_field(:y, "solid_fuel", type: :quantitative)
+```
+
+```elixir
+# With the shorthand api
+Vl.new(title: "Fuels", width: 400)
+|> Data.chart(fuels, [type: :line, point: true], x: "total", y: "solid_fuel")
+```
+
+If a channel requires more configuration, the flexibility of the API comes into play.
+
+```elixir
+Vl.new(width: 500, height: 300, title: "Fuels")
+|> Vl.data_from_values(fuels, only: ["total", "solid_fuel"])
+|> Vl.mark(:point)
+|> Vl.encode_field(:x, "total", type: :quantitative)
+|> Vl.encode_field(:y, "solid_fuel", type: :quantitative)
+|> Vl.encode_field(:color, "total", type: :quantitative, scale: [scheme: "category10"])
+```
+
+In the example above, we have a color channel that requires more customization. While it's possible to get the exact same plot using only the shorthand API, the expressiveness may be sacrificed. It's precisely in these cases that using the API together with the main module will probably result in more readable code. Let's take a look and compare the possible combinations between the API and the `VegaLite` module.
+
+```elixir
+# Using mainly the shorthand api
+Vl.new(width: 500, height: 300, title: "Combined")
+|> Data.chart(fuels, :point,
+  x: "total",
+  y: "solid_fuel",
+  color: [field: "total", type: :quantitative, scale: [scheme: "category10"]]
+)
+```
+
+```elixir
+# Piping the shorthand api into a enconde_field
+Vl.new(width: 500, height: 300, title: "Fuels")
+|> Data.chart(fuels, :point, x: "total", y: "solid_fuel")
+|> Vl.encode_field(:color, "total", type: :quantitative, scale: [scheme: "category10"])
```

Comment:
There is an issue in this approach. This code only works because both `x` and `color` use "total". But if the color was encoded with another value, then its field would not be included in `only`. Thoughts?

Reply:
Good catch! We rely on `Table.Reader`, so it might be better to remove the implicit `only` and allow passing it as an option of the `data` argument. Or maybe make it implicit by default and use the option to override it. 🤔

Reply:
I think we can have one option called additional_fields or something?

Reply:
I prefer `only` for consistency, like in `values_from`

Reply:
Data.chart([data: fuels, only: …], :point, x: "total", y: "solid_fuel")
And that should be the only argument accepted for data

Reply:
Also, you said nothing about having or not a default…

Reply:
We should have a default, for sure. But I was wondering if we could have that as part of fields somehow:

    Data.chart(fuels, :point, x: "total", y: "solid_fuel", additional_fields: [:foo, :bar])

Another idea is to move the logic `VegaLite` itself. We could support `only: :lazy` and then we compute it the time we build the JSON.

Reply:
Ok!